client/src/controllers/LinkController.ts

```diff
@@ -57,12 +59,14 @@ export class LinkController extends Controller<HTMLElement> {
     this.element.setAttribute(this.attrNameValue, currentUrl.toString());
   }
 
-  setParamsFromSwapRequest(e: CustomEvent<{ requestUrl?: string }>) {
+  setParams(e?: CustomEvent<{ requestUrl?: string }>) {
```

Comment:
Nice, this is a much cleaner approach.

Is it worth supporting params while we are changing things?

Also, a personal preference, but I find using the variable name `event` is much clearer, we have linting rules for avoiding short names but I think it allows `e`.

I know there is a JS convention sometimes for `e` but this can be many things, lots of our other Stimulus controllers and newer code uses `event` for when the variable is an event.

```suggestion
  setParams(event?: CustomEvent<{ requestUrl?: string }> { params?: { requestUrl?: string }}) {
      # ... 
      const { requestUrl } = {...event.detail, event.params };
```